### ISSUE TITLE
timediv() doesn't seem to work as advertised (or maybe it does ;-)

### ISSUE THREAD

**Issue by totalgee** [tmhglnd/mercury-playground#42]
I have found that in some cases `timediv` doesn't do what (I thought) it's supposed to do. In this example, it sounds more like it's playing 3 and then 2 beats, rather than 4 and then 1 beats:

```
new sample kick_909 time(1/4) play([1 0]) timediv([4 1])
```

Another confusing example (now it sounds like 3, 1, 2, 1 with some syncopation/offset):

```
new sample snare_909 time(1/4) play([1 0 1 0]) timediv([4 1 1 1])
```

I would have expected the first beat to play four times (and 4x faster), then the others to play once at normal speed.

It may be that I misunderstand what it's supposed to do (edit: yes), but if you use it without a `play()` list then it behaves as I'd expect.

Hmm, I guess the tutorial says it steps through *the other lists*, so I guess that's what's happening here, it's also stepping quickly through the `play` list. But it's not what I would have expected/wanted. Perhaps what I want is to `ratchet` (but this is not implemented in the web playground yet)...?

**Comment by totalgee** [comment]
Also, the documentation is missing for `timediv` and `ratchet`. (-;

**Comment by tmhglnd** [comment]
Yes, I agree `timediv()` needs some improvements here. I agree that the behaviour should be more like a ratcheting effect indeed, meaning that `time(1/4) play([1 1 0])` combined with `timediv([4 1 1])` should result in 4 16th-notes then one quarter-note then a quarter-rest.

**Comment by tmhglnd** [comment]
I have worked on this bug a bit. For now i've been able to get the "ratchet-counter" to repeat for every list except the `play()`. 

So for example:

```js
new synth sine note([0 7 12] 2) time(1/4) timediv([1 2 3])
```

results in 1x 0, 2x 7, 3x 12 in a 1/4, 1/8 and 1/12 rhythm.

For the issue with `play([1 0])` it is more tricky to fix, I need more time to look at what the options are. This is because the `time()` increments the steps in `play()` and `play()` increments the steps in `timediv()`, so it's a bit too tangled up at the moment.
